README.md

```diff
@@ -1,13 +1,219 @@
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0)
+[![Go Report Card](https://goreportcard.com/badge/github.com/Mellanox/rdma-cni)](https://goreportcard.com/report/github.com/Mellanox/rdma-cni)
+[![Build Status](https://travis-ci.com/Mellanox/rdma-cni.svg?branch=master)](https://travis-ci.com/Mellanox/rdma-cni)
+[![Coverage Status](https://coveralls.io/repos/github/Mellanox/rdma-cni/badge.svg)](https://coveralls.io/github/Mellanox/rdma-cni)
+
 # RDMA CNI plugin
-CNI compliant plugin for RDMA namespace aware network interfaces
+CNI compliant plugin for network namespace aware RDMA interfaces.
+
+RDMA CNI plugin allows network namespace isolation for RDMA workloads in a containerized environment.
 
 # Overview
-This software will enable namespace isoloation for RDMA workloads on a kubernetes
-cluster.
+RDMA CNI plugin is intended to be run as a chained CNI plugin (introduced in [CNI Specifications `v0.3.0`](https://github.com/containernetworking/cni/blob/v0.3.0/SPEC.md#network-configuration)).
+It ensures isolation of RDMA traffic from other workloads in the system by moving the associated RDMA interfaces of the
+provided network interface to the container's network namespace path.
+
+The main use-case (for now...) is for containerized SR-IOV workloads orchestrated by [Kubernetes](https://kubernetes.io/)
+that perform [RDMA](https://community.mellanox.com/s/article/what-is-rdma-x) and wish to  leverage network namespace
+isolation of RDMA devices introduced in linux kernel `5.2.0`.
+
+# Requirements
+
+## Hardware
+SR-IOV capable NIC which supports RDMA.
+
+### Supported Hardware
+
```

Comment:
need add support  FW latest GA and above

Reply:
why is that needed ? because of what we test with ? rdma namespace aware does not depend on the NIC Firmware to my knowledge.

Reply:
Ok dont add it